File: trab1/python/precisao.py
# coding=utf-8
from math import log, floor

p = 1.0
while p + 1 > 1:
    p = p / 2
p = 2 * p
d = abs(floor(log(p, 10)))
n = floor(abs(log(p, 2)))
print 'Precisão: %g' % p
print 'Núm. dígitos significativos decimais: %i' % d
print 'Núm. bits da mantissa: %i' % n

File: erlang/trab1/precisao.erl
-module(precisao).
-export([main/0]).

precisao(1.0, P) -> P;
precisao(S, _) -> precisao((S + 1) / 2, S - 1).

main() ->
    P = precisao(2,0),
    D = abs(trunc(math:log10(P))),
    N = trunc(abs(math:log(P)/math:log(2))),
    io:fwrite("~ts: ~g~n", [<<"Precisão">>, P]),
    io:fwrite("~ts: ~b~n", [<<"Núm. dígitos significativos decimais">>, D]),
    io:fwrite("~ts: ~b~n", [<<"Núm. bits da mantissa">>, N]).

